src/mlpack/methods/random_forest/random_forest_regressor.hpp

```diff
@@ -0,0 +1,401 @@
+#include "bootstrap.hpp"
+
+namespace mlpack {
+
+/**
+ * This class implements a random forest regressor.
+ * The RandomForestRegressor class implements a parallelized random forest regressor 
+ * that supports numerical and categorical features, by default using MSE (minimum 
+ * sqaured error) to chosee which feature to split on. Random forests are a collection 
+ * of decision trees that give better performance than a single decision tree.
+ */
+
+/*
+ * FitnessFunction : the measure of goodness to use when deciding on tree splits.
+ * DimensionSelectionType: the strategy used for proposing dimensions to attempt to split on.
+ * NumericSplitType: the strategy used for finding splits on numeric data dimensions.
+ * CategoricalSplitType: the strategy used for finding splits on categorical data dimensions.
+ * UseBootstrap: a boolean indicating whether or not to use a bootstrap sample when training 
+ *               each tree in the forest.
+*/
+
+template<typename FitnessFunction = MSEGain,
+         typename DimensionSelectionType = MultipleRandomDimensionSelect,
+         template<typename> class NumericSplitType = BestBinaryNumericSplit,
+         template<typename> class CategoricalSplitType = AllCategoricalSplit,
+         bool UseBootstrap = true>
+class RandomForestRegressor
+{
+ public:
+  //! Allow access to the underlying decision tree type.
+  typedef DecisionTreeRegressor<FitnessFunction, NumericSplitType, CategoricalSplitType,
+      DimensionSelectionType> DecisionTreeType;
+
+  /**
+   * Construct the random forest without any training or specifying the number
+   * of trees.  Predict() will throw an exception until Train() is called.
+   */
+  RandomForestRegressor();
+
+  /**
+   * Create a random forest, training on the given data and responses with
+   * the given number of trees.  The minimumLeafSize and minimumGainSplit
+   * parameters are given to each individual decision tree during tree building.
+   * Optionally, you may specify a DimensionSelectionType to set parameters for
+   * the strategy used to choose dimensions.
+   *
+   * @param dataset Dataset to train on.
+   * @param responses Responses for each training point.
+   * @param numTrees Number of trees in the forest.
+   * @param minimumLeafSize Minimum number of points in each tree's leaf nodes.
+   * @param minimumGainSplit Minimum gain for splitting a decision tree node.
+   * @param maximumDepth Maximum depth for the tree.
+   * @param dimensionSelector Instantiated dimension selection policy.
+   */
+  template<typename MatType, typename ResponsesType>
+  RandomForestRegressor(const MatType& dataset,
+               const arma::Row<ResponsesType>& responses,
+               const size_t numTrees = 20,
+               const size_t minimumLeafSize = 1,
+               const double minimumGainSplit = 1e-7,
+               const size_t maximumDepth = 0,
+               DimensionSelectionType dimensionSelector =
+                   DimensionSelectionType());
```

Comment:
```suggestion
                        const arma::Row<ResponsesType>& responses,
                        const size_t numTrees = 20,
                        const size_t minimumLeafSize = 1,
                        const double minimumGainSplit = 1e-7,
                        const size_t maximumDepth = 0,
                        DimensionSelectionType dimensionSelector =
                            DimensionSelectionType());
```
Just to make everything align.  Can you do that to the other places in this file too?